spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/RewriteDataFilesSparkAction.java

```diff
@@ -507,4 +645,54 @@ public int totalGroupCount() {
       return totalGroupCount;
     }
   }
+
+  private static class MakeDeleteFile implements MapFunction<Row, DeleteFile> {
```

Comment:
@zinking brought up a good point in #6581 that this may not be necessary if we can remove a delete file by path.   It can probably reduce memory pressure and allow us not to project unnecessary fields , for example  fileSize, here.

@dramaticlly pointed out that the call seems not to work for delete files.  ref: https://github.com/apache/iceberg/blob/apache-iceberg-1.4.3/core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java#L205

@aokolnychyi any thoughts here?

Reply:
Replied [here]. I think we use use `SparkDeleteFile` to wrap Spark rows as delete files.